extensions/resteasy-reactive/quarkus-resteasy-reactive-kotlin-serialization/runtime/src/main/kotlin/io/quarkus/kotlin/serialization/KotlinSerializationMessageBodyWriter.kt

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.kotlin.serialization
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.serializer
+import org.jboss.resteasy.reactive.common.providers.serialisers.JsonMessageBodyWriterUtil
+import org.jboss.resteasy.reactive.server.providers.serialisers.json.JsonMessageServerBodyWriterUtil
+import org.jboss.resteasy.reactive.server.spi.ServerMessageBodyWriter.AllWriteableMessageBodyWriter
+import org.jboss.resteasy.reactive.server.spi.ServerRequestContext
+import java.io.OutputStream
+import java.lang.reflect.Type
+import javax.inject.Inject
+import javax.ws.rs.Produces
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.MultivaluedMap
+
+@Produces("application/json", "application/*+json", "text/json")
+@OptIn(ExperimentalSerializationApi::class)
+class KotlinSerializationMessageBodyWriter(@Inject var json: Json) : AllWriteableMessageBodyWriter() {
+    override fun writeTo(
+        o: Any, type: Class<*>, genericType: Type, annotations: Array<Annotation>, mediaType: MediaType,
+        httpHeaders: MultivaluedMap<String, Any>, entityStream: OutputStream
+    ) {
+        JsonMessageBodyWriterUtil.setContentTypeIfNecessary(httpHeaders)
+        if (o is String) { // YUK: done in order to avoid adding extra quotes...
+            entityStream.write(o.toByteArray())
+        } else {
+            entityStream.write(json.encodeToString(o).toByteArray())
```

Comment:
I would suggest using the new enncodeToStream and decodeFromStream methods?

Reply:
omg.  I don't know why it's that way.  After a while you stop seeing details, I guess.  Will fix.  elsewhere uses `toStream`.